### ISSUE TITLE
[TASK] Migrate various names to conform KubeVirt conventions

### ISSUE THREAD

**Issue by starbops** [starbops/kubevirtbmc#5]
Due to the upcoming [project migration schedule](https://github.com/kubevirt/community/blob/main/design-proposals/kubevirtbmc.md), we have some tasks that need to be done before the migration to ensure smooth progress.

One of the tasks is to renaming the CRD and change the API group of it to conform the KubeVirt convention:

- Changing from `KubeBMC` to `VirtualMachineBMC`
- Changing from `zespre.com` to `kubevirt.org`

Also, the components `kubebmc-controller` and `kbmc` should be renamed to `virt-bmc-controller` and `virt-bmc`, respectively.

Finally, the project's name will be changed to `KubeVirtBMC` to convey that this project is targeted to work with KubeVirt.

P.S. This is a follow-up issue of #1.

**Comment by kashifest** [comment]
I can help in this one 
/assign @kashifest

**Comment by starbops** [comment]
@kashifest 
Thank you for volunteering to handle this issue! I really appreciate your help. I will assign this issue to you now. If you need any further information or assistance, please let me know.
